state/s3.yaml

```diff
@@ -227,6 +322,52 @@ Resources:
           Condition:
             StringEquals:
               'aws:sourceVpce': {'Fn::ImportValue': !Sub '${ParentVpcEndpointStack}-EndpointS3'}
+  BucketPolicyFlowLogWrite:
+    Condition: HasFlowLogWriteAccess
+    Type: 'AWS::S3::BucketPolicy'
+    Properties:
+      Bucket: !Ref Bucket
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement: # https://docs.aws.amazon.com/vpc/latest/userguide/flow-logs-s3.html#flow-logs-s3-permissions
+        - Sid: AWSLogDeliveryWrite
+          Effect: Allow
+          Principal:
+            Service: 'delivery.logs.amazonaws.com'
+          Action: 's3:PutObject'
+          Resource: !Sub '${Bucket.Arn}/*AWSLogs/${AWS::AccountId}/*'
+          Condition:
+            StringEquals:
+              's3:x-amz-acl': 'bucket-owner-full-control'
+        - Sid: AWSLogDeliveryAclCheck
+          Effect: Allow
+          Principal:
+            Service: 'delivery.logs.amazonaws.com'
+          Action: 's3:GetBucketAcl'
+          Resource: !GetAtt 'Bucket.Arn'
+  BucketPolicyFlowLogCrossAccountWrite:
```

Comment:
This allows any AWS account to publish their flow logs to your AWS account, correct? Is there no way to restrict based on OU/Account Id/...?

Reply:
I made an experiment with the condition `aws:PrincipalOrgID` but it doesn't seem to work. Let me open a support case and see what AWS suggests...

Reply:
removed